lib/interceptor.js

```diff
@@ -334,6 +335,10 @@ Interceptor.prototype.match = function match(options, body, hostNameOnly) {
             } else if (expVal instanceof RegExp) {
               isMatch = common.matchStringOrRegexp(val, expVal)
             } else if (_.isArray(expVal) || _.isObject(expVal)) {
+              // TODO: better array comparison. Order shouldn't matter.
+              // The `qs` lib groups duplicate keys into an array of values, the current
+              // implementation will not match: `/path?foo=one&foo=two` - `/path?foo=two&foo=one`
```

Comment:
As there's nothing in the spec that says those should be treated the same, I would argue those are two different queries.

Reply:
You're absolutely correct. 
Even the spec for [`URLSearchParams.getAll`](https://url.spec.whatwg.org/#dom-urlsearchparams-getall) specifies that duplicate keys must maintain order.
Will remove.